packages/camera/example/lib/main.dart

```diff
@@ -24,149 +26,316 @@ IconData cameraLensIcon(CameraLensDirection direction) {
+  }
+
+  Future<String> startVideoRecording() async {
+    if (!controller.value.isInitialized) {
+      showInSnackBar('Error: select a camera first.');
+      return null;
+    }
+    final Directory extDir = await getApplicationDocumentsDirectory();
+    final String dirPath = '${extDir.path}/Movies/flutter_test';
+    await new Directory(dirPath).create(recursive: true);
+    final String filePath = '$dirPath/${timestamp()}.mp4';
+    try {
+      videoPath = filePath;
+      await controller.startVideoRecording(filePath);
+    } on CameraException catch (e) {
+      logError(e.code, e.description);
+    }
+    return filePath;
+  }
+
+  Future<void> stopVideoRecording() async {
+    if (controller.value.isRecordingVideo) {
+      try {
+        await controller.stopVideoRecording();
+      } on CameraException catch (e) {
+        logError(e.code, e.description);
       }
-      final String tempPath = tempDir.path;
-      final String path = '$tempPath/picture${pictureCount++}.jpg';
-      await controller.capture(path);
+      final VideoPlayerController vcontroller =
+          new VideoPlayerController.network('file://$videoPath');
+      vcontroller.play();
+      vcontroller.setLooping(true);
+      videoPlayerListener = () {
+        if (videoController != null && videoController.value.size != null) {
+          videoController.removeListener(videoPlayerListener);
+          // Refreshing the state to update video player with the correct ratio.
+          if (mounted) setState(() {});
+        }
+      };
+      vcontroller.addListener(videoPlayerListener);
+      await vcontroller.initialize();
       if (!mounted) {
-        return;
+        return null;
       }
-      setState(
-        () {
-          imagePath = path;
-        },
-      );
+      setState(() {
+        imagePath = null;
+        videoController?.dispose();
+        videoController = vcontroller;
+      });
     }
   }
+
+  Future<String> takePicture() async {
+    if (!controller.value.isInitialized) {
+      showInSnackBar('Error: select a camera first.');
+      return null;
+    }
+    final Directory extDir = await getApplicationDocumentsDirectory();
+    final String dirPath = '${extDir.path}/Pictures/flutter_test';
+    await new Directory(dirPath).create(recursive: true);
+    final String filePath = '$dirPath/${timestamp()}.jpg';
+    try {
+      await controller.takePicture(filePath);
+    } on CameraException catch (e) {
+      logError(e.code, e.description);
+    }
+    return filePath;
+  }
+}
+
+class CameraApp extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return new MaterialApp(
+      home: new CameraExampleHome(),
+    );
+  }
 }
 
 List<CameraDescription> cameras;
 
 Future<Null> main() async {
-  cameras = await availableCameras();
-  runApp(new MaterialApp(home: new CameraExampleHome()));
+  // Save the available cameras in this variable first before initializing the app.
```

Comment:
// Fetch the available cameras before initializing the app.

Reply:
Done